sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -452,12 +483,75 @@ edges.
 
 The `node` class provides the {crs}[common reference semantics].
 
+==== Dynamic Parameters
+
+Dynamic parameters are inputs to a node's command-group which can be updated by
+the user after the node has been added to a graph. They are updated through an
+executable graph which contains the node associated with a dynamic parameter
+object.
+
+Dynamic parameters may represent either a `sycl::buffer` or a pointer to a USM
+allocation. When dynamic parameters are updated the new value must be
+compatible with the existing value. For USM allocations this must be a pointer
+to a valid USM allocation in the same context, and for buffers the new value
+must have a total memory size that is greater than or equal to the original
+value.
+
+Dynamic parameters can optionally have a user-specified string label associated
+with them. This can provide meaningful names to help track what the parameters
+are when updating them. These labels are not used by the SYCL runtime in any
+way.
+
+
+==== Node Member Functions
+
+Table {counter: tableNumber}. Member functions of the `node` class.
+[cols="2a,a"]
+|===
+|Member Function|Description
+
+|
 [source,c++]
 ----
-namespace sycl::ext::oneapi::experimental {
-  class node {};
-}
+node_type getType() const;
+----
+|Returns a value representing the type of command this node represents.
+
+|
+[source,c++]
+----
+std::vector<Node> getDependencies() const;
 ----
+|Returns a list of the nodes which this node depends on.
+
+|
+[source,c++]
+----
+template <typename T>
+dynamic_parameter registerDynamicParameter(T param, std::string label = "") const;
```

Comment:
Do I understand this proposal correctly that a recorded graph can be queried with `getNodes()` and the returned nodes can then be updated by calling `registerDynamicParameter()`? An example for the update using the explicit API and record & replay might be beneficial.

Reply:
I agree an example of this would probably useful, though I think we're in danger of the spec becoming unreadably long. Perhaps it is worth exploring alternative options for more detailed usage examples, perhaps a separate document or examples repo?

Reply:
In general I agree with Ben we long term we need an examples repo or something separate for this kind of reference, but think in the short term this feature is complicated/important enough to warrant this, so will add at least one example or registration & update.